From a3bc2fcc563fe7f7b8ae70dc01686f98862d3c1a Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 17 Mar 2021 15:30:52 +0200
Subject: [PATCH] [SW-34119] preboot: uart: switching to fast run execution

Since during the cmd_module execution, uart is running in buffered
mode, it can be executed as fast running operation.

Change-Id: I9b2d20e16dc819b96d31a76fba039411576d4459
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/module_uart.c b/preboot/module_uart.c
index aaae9a1..3f4d7ee 100644
--- a/preboot/module_uart.c
+++ b/preboot/module_uart.c
@@ -15,7 +15,7 @@
 	    (preboot_uart_tx_full() || preboot_uart_rbuf_empty()))
 		return NOT_FOUND;
 
-	return FOUND_SLOW_RETURN;
+	return FOUND_FAST_RETURN;
 }
 
 __weak void hl_esc_key_handle(void) {}
diff --git a/preboot/preboot.c b/preboot/preboot.c
index fed02e5..112c08b 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -68,9 +68,9 @@
 		.setup = module_uart_setup,
 		.check = module_uart_check,
 		.disable = noop,
-		.run = module_uart_run,
+		.run = noop,
 		.enable = noop,
-		.run_fast = noop,
+		.run_fast = module_uart_run,
 		.teardown = module_uart_teardown,
 		.run_noreturn = noop,
 	},
